### ISSUE TITLE
Activity webhook

### ISSUE THREAD

**Issue by rentalhost** [Bungie-net/api#418]
Could be possible at some time the Bungie API supports an activity webhook? Eg. when an user completes some activity a script on my server be immediatelly reported with the carnage report from this activity. 

It should be installed by user without user permission (once that is a public data). I think you can limit the activity hook to about 1K users by app. Users could be included or revoked anytime.

It will add a lot of advantages to clan bots, once that we don't need anymore search deeply by recent users avtivities.

**Comment by floatingatoll** [comment]
What’s a specific example advantage that would you use webhooks provide to players as a site operator, that are not available today with the 30 second refresh model?

**Comment by vthornheart-bng** [comment]
It's an interesting idea!  It's unfortunately not something we're set up to provide at this time, but I can see where that would be appealing vs. the current requirement of periodically polling to get that data.  I'm going to flag it as being on the distant future wishlist; while I'm pretty sure it's not something we'll be able to get to in the near future, I do think it'd be an interesting idea to explore in future versions of the API!

**Comment by unisys12** [comment]
Just to give another example of how this might be something devs in the community could use and expanding a little on the example @rentalhost gave - 

I wrote a Chrome extension near the end of D1 _(my timing was not exactly on point)_ that pulled Grimoire information related to the activity your character entered and displayed it. For example, if you were running Crucible and started to fly into a match, the extension would display the name and image of the map that you were about to fly into, along with any Grimoire Cards related to that map. Same for strikes, story missions and raids. Although I limited the polling to every 10secs _(gave the best instant display of info as possible)_, there was a ton of HTTP requests flying by. Addy Osmani would not have been proud of me.

I will admit that my example is pretty extreme, given that it covered way more than a single endpoint. But if you combined the flexibility of the current component system with a webhook that one could monitor, it would be pretty cool. I will also admit that this is something that a dev could fairly easily build for themselves too, with a little bit of research, so take all this with a grain of salt and call it food for thought.

**Comment by vthornheart-bng** [comment]
Yeah, I think that is definitely a strong argument for having a webhook.  The real holdup for us is that we're not set up well for it at the moment, and the cost-to-benefit ratio is unfortunately very high, so it's not something we're likely to see get put in soon.  But I do hope it is something we get to, even if for theoretical future versions of Destiny, because I could definitely see scenarios like that being useful!
